src/main/assisted-installer-controller/assisted_installer_main.go

```diff
@@ -69,3 +80,26 @@ func main() {
 	logger.Infof("Waiting fo all go routines to finish")
 	wg.Wait()
 }
+
+// ProxyFromEnvVars provides an alternative to http.ProxyFromEnvironment since it is being initialized only
+// once and that happens by k8s before proxy settings was obtained. While this is no issue for k8s, it prevents
+// any out-of-cluster traffic from using the proxy
+func ProxyFromEnvVars(req *http.Request) (*url.URL, error) {
```

Comment:
not sure we need all 3 one line funcs? can't we move all to utils and call it CreateHttpProxyFunc .... ?All here is proxy oriented and no need to call one without calling others. So not sure what logic we split here?

Reply:
That's because https://github.com/openshift/assisted-installer/blob/master/src/inventory_client/inventory_client.go#L76 expects to recieve a function as value for Proxy:
```
	transport := requestid.Transport(&http.Transport{
		Proxy: http.ProxyFromEnvironment, // this one is a function
		DialContext: (&net.Dialer{
...
``
it should be possible to inline one of the 3 methods.